src/main/java/uk/gov/hmcts/darts/audio/controller/AudioController.java

```diff
@@ -61,11 +78,28 @@ public ResponseEntity<byte[]> preview(Integer mediaId, String httpRangeList) {
         return StreamingResponseEntityUtil.createResponseEntity(audioMediaFile, httpRangeList);
     }
 
-    @Override
+
+    @RequestMapping(
+        method = RequestMethod.GET,
+        value = "/audio/preview/{media_id}",
+        produces = {"text/event-stream", "application/json+problem"}
+    )
+    @SneakyThrows
     @SecurityRequirement(name = SECURITY_SCHEMES_BEARER_AUTH)
-    public ResponseEntity<Void> addAudio(MultipartFile file, AddAudioMetadataRequest metadata) {
-        audioService.addAudio(file, metadata);
-        return new ResponseEntity<>(HttpStatus.OK);
+    @Authorisation(contextId = MEDIA_ID,
+        securityRoles = {JUDGE, REQUESTER, APPROVER, TRANSCRIBER, TRANSLATION_QA, RCJ_APPEALS},
+        globalAccessSecurityRoles = {JUDGE})
+    public Flux<ResponseEntity<byte[]>> previewAlternative(
```

Comment:
Also sorry I'm naive to the specific implemenation of flux. Spring provides a flux RestController annotation as part of the dependnecy spring-boot-starter-webflux. Do we not need that instead of the MVC one?

Reply:
Did we try using https://www.baeldung.com/spring-deferred-result? DId that help with the timeout problem. Conscious that this allows us to come away from a react approach

Reply:
> Also sorry I'm naive to the specific implemenation of flux. Spring provides a flux RestController annotation as part of the dependnecy spring-boot-starter-webflux. Do we not need that instead of the MVC one?

Can you show where the Webflux rest controller is located I can't seem to find it. I also suspect in that case the annotation would apply to the whole controller which is what we don't want.

Reply:
> Did we try using https://www.baeldung.com/spring-deferred-result? DId that help with the timeout problem. Conscious that this allows us to come away from a react approach

I don't think this will help as we not looking for a non blocking call but a call that will constantly return data to stop the firewall from closing the connection.

Reply:
I did read about web flux schedulers to avoid such a situation https://www.baeldung.com/spring-webflux-concurrency